compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -441,12 +441,17 @@ impl<'a, Ty> TyAndLayout<'a, Ty> {
                         let mut total = start;
 
                         for i in 0..layout.fields.count() {
+                            let field = layout.field(cx, i);
+                            if field.is_zst() {
```

Comment:
I'm not sure if this should ignore all ZST or only all 1-ZST. That depends on whether alignment of a struct affects whether it is considered to be a homogeneous aggregate.

Reply:
@bjorn3 do you know what would be more appropriate here?

Reply:
I think only ignoring 1-ZSTs here is safer. I don't know if there are any abi's where it will matter though.

Reply:
Let's do the safe thing then.

Reply:
FWIW for `Abi::Scalar/ScalarPair`, we ignore all ZST, but not entirely -- there is a 2nd pass checking their alignment and falling back to Aggregate ABI if the alignment is too high. That way `(i32, [u16; 0])`  can get Scalar ABI (compatible with `i32`) but `(i16, [u32; 0])` is an Aggregate (the highly-aligned ZST prevents making this `i16`-compatible).